File: tests/go/hello.go
package main

import (
	"fmt"
	"os"
)

func main() {
	// Verify non-root execution
	uid := os.Getuid()
	if uid == 0 {
		fmt.Fprintf(os.Stderr, "Container must not run as root (got uid=%d)\n", uid)
		os.Exit(1)
	}

	fmt.Printf("Running as uid: %d\n", uid)
	fmt.Println("All smoke test assertions passed.")
}

File: tests/typescript/hello.js
"use strict";

const process = require("process");

// Verify non-root execution
const uid = process.getuid();
if (uid === 0) {
  console.error(`Container must not run as root (got uid=${uid})`);
  process.exit(1);
}

// Verify Node version matches expected (if provided via env)
const expected = process.env.EXPECTED_VERSION;
if (expected) {
  const actual = process.version.replace(/^v/, "");
  if (actual !== expected) {
    console.error(`Version mismatch: expected ${expected}, got ${actual}`);
    process.exit(1);
  }
}

console.log(`Node version: ${process.version}`);
console.log(`Running as uid: ${uid}`);
console.log("All smoke test assertions passed.");
